Start with importing the pandas library which will allow us to read the csv file. We use the import to add the pandas module

In [1]:
import pandas as pd

Need to load the csv into the pandas dataframe. Pulling from my GitHub repository. Define the variable as df and use the read_csv() function to load the csv into pandas. I added the extra df line to display the data

In [21]:
df = pd.read_csv("https://github.com/KcirtapdiaZ/IS_362_Data_Acquisition_and_Mgmt/raw/main/Week4-Project1.csv")
df

Airline    Destination   Status  Count
0    Alaska    Los Angeles  On Time    497
1    Alaska    Los Angeles  Delayed     62
2    Alaska        Pheonix  On Time    221
3    Alaska        Pheonix  Delayed     12
4    Alaska      San Diego  On Time    212
5    Alaska      San Diego  Delayed     20
6    Alaska  San Francisco  On Time   1841
7    Alaska  San Francisco  Delayed    305
8    Alaska        Seattle  On Time    694
9    Alaska        Seattle  Delayed    117
10  AM West    Los Angeles  On Time    694
11  AM West    Los Angeles  Delayed    117
12  AM West        Pheonix  On Time   4840
13  AM West        Pheonix  Delayed    415
14  AM West      San Diego  On Time    383
15  AM West      San Diego  Delayed     65
16  AM West  San Francisco  On Time    320
17  AM West  San Francisco  Delayed    129
18  AM West        Seattle  On Time    201
19  AM West        Seattle  Delayed     61

Filter the dataset to show only delayed flights to make it easier to calculate the delays for each airline. This creates a new DataFrame that only contains the delayed flights. To simplify later code I filter now

In [16]:
delays = df[df["Status"] == "Delayed"]
delays

Airline    Destination   Status  Count
1    Alaska    Los Angeles  Delayed     62
3    Alaska        Pheonix  Delayed     12
5    Alaska      San Diego  Delayed     20
7    Alaska  San Francisco  Delayed    305
9    Alaska        Seattle  Delayed    117
11  AM West    Los Angeles  Delayed    117
13  AM West        Pheonix  Delayed    415
15  AM West      San Diego  Delayed     65
17  AM West  San Francisco  Delayed    129
19  AM West        Seattle  Delayed     61

In order to calculate the total delays by airline, need to group the delayed flights by each airline. This is make it easier to compare which airline has the most delays overall. Best way to add up the number of delays is to group. By using groupby we can combine the data by category

In [17]:
delays_by_airline = delays.groupby("Airline")["Count"].sum()
delays_by_airline

Airline
AM West    787
Alaska     516
Name: Count, dtype: int64

To calculate the overall delay percentage we need to first calculate the total number of flights for each airline. Both On Time and Delayed. I used groupby again for code consistancy

In [18]:
total_flights = df.groupby("Airline")["Count"].sum()
total_flights

Airline
AM West    7225
Alaska     3981
Name: Count, dtype: int64

We can now calculate the delay percentage of each airline. We divide variable "delays_by_airline" by "total_flights" and multiply by 100.

In [19]:
delay_percentages = (delays_by_airline / total_flights) * 100
delay_percentages

Airline
AM West    10.892734
Alaska     12.961567
Name: Count, dtype: float64

Analyzing the data of the delays between each airline, the percentages are a lot closer than you may think just by baseing your conclusion off of which airline had more total flights or more delays. The percentage shows you a better picture, becasue although AM West had more delays at 787 compared to 516, they also had substancially more total flights at 7225 compared to 3981. Because the delay percentage is so close maybe factors like destination, flight length, or type of plane used should be factored in to determine a true airline reliability. If AM West didnt have 2300 more flights the delay percentage may be higher which would reverse the conclusion of AM West is more reliable because they have less delay percentage.